sklearn/decomposition/kernel_pca.py

```diff
@@ -259,7 +262,7 @@ def inverse_transform(self, X):
         "Learning to Find Pre-Images", G BakIr et al, 2004.
         """
         if not self.fit_inverse_transform:
-            raise ValueError("Inverse transform was not fitted!")
+            raise NotFittedError("Inverse transform was not fitted.")
```

Comment:
I don't know what its purpose would be programmatically; I think this isn't Java and having a specialised exception requires a specialised usecase.